perl/shop/t/Basket.t

```diff
@@ -1,319 +1,364 @@
+    is( $hResultDelete->{'deleted'}, 1,             'testDeleteLineItem: deleted' );
+}
 
-    my $hLineItemContainer  = $hInfo->{'LineItemContainer'};
-    my $hLineItemContainer2 = $basketRef->{'LineItemContainer'};
-    ok( $hLineItemContainer->{'TaxArea'}    eq $hLineItemContainer2->{'TaxArea'},       'tax area' );
-    ok( $hLineItemContainer->{'TaxModel'}   eq $hLineItemContainer2->{'TaxModel'},      'tax model' );
-    ok( $hLineItemContainer->{'CurrencyID'} eq $hLineItemContainer2->{'CurrencyID'},    'currencyid' );
+sub testExistsBasket {
+    my $BasketPath = _setupTestBasketInDB();
 
-    ok( $hLineItemContainer->{'ProductLineItems'}, 'list of all product line items');
-    my @ProductLineItems =
-        map {{$_->{'Product'}, { 'Quantity' => $_->{'Quantity'}, 'OrderUnit' => $_->{'OrderUnit'}}}} sort {$a->{Product} le $b->{Product}}
-        @{$hLineItemContainer->{'ProductLineItems'}};
-    my @ProductLineItemsRef =
-        map {{$_->{'Product'},{ 'Quantity' => $_->{'Quantity'}, 'OrderUnit' => $_->{'OrderUnit'}}}} sort {$a->{Product} le $b->{Product}}
-        @{$hLineItemContainer2->{'ProductLineItems'}};
+    my $ahResultExists = $BasketService->exists( [$BasketPath] )->result;
+    is( scalar @$ahResultExists, 1, 'testExistsBasket: response format' );
 
-    is_deeply( \@ProductLineItems, \@ProductLineItemsRef, 'product line items');
+    my $hResultExists = $ahResultExists->[0];
+    ok( !$hResultExists->{'Error'}, 'testExistsBasket: check error' );
+    diag $hResultExists->{'Error'}->{'Message'} . "\n" if $hResultExists->{'Error'};
 
-    return $hInfo;
+    is( $hResultExists->{'Path'},   $BasketPath, 'testExistsBasket: basket path' );
+    is( $hResultExists->{'exists'}, 1,           'testExistsBasket: exists' );
 }
 
 sub testDeleteBasket {
-    my ( $basketPath ) = @_;
+    my $BasketPath = _setupTestBasketInDB();
 
-    my $ahResults = $BasketService->delete( [$basketPath] )->result;
-    is( scalar @$ahResults, 1, 'delete: result count' );
+    my $ahResultDelete = $BasketService->delete( [$BasketPath] )->result;
+    is( scalar @$ahResultDelete, 1, 'testDeleteBasket: result count' );
 
-    my $hDelete = $ahResults->[0];
-    ok( !$hDelete->{'Error'}, 'delete: no error' );
-    diag $hDelete->{'Error'}->{'Message'}."\n" if $hDelete->{'Error'};
+    my $hResultDelete = $ahResultDelete->[0];
+    ok( !$hResultDelete->{'Error'}, 'testDeleteBasket: check error' );
+    diag $hResultDelete->{'Error'}->{'Message'} . "\n" if $hResultDelete->{'Error'};
 
-    ok( $hDelete->{'Path'} eq $basketPath, 'delete: order path' );
-    is( $hDelete->{'deleted'}, 1, 'deleted?' );
+    is( $hResultDelete->{'Path'},    $BasketPath, 'testDeleteBasket: basket path' );
+    is( $hResultDelete->{'deleted'}, 1,           'testDeleteBasket: deleted' );
 }
 
-sub testUpdateBasket {
-    my ($Path, $Basket) = @_;
-    $Basket->{Path} = $Path;
+sub testaddProductLineItem {
```

Comment:
Maybe add call to `getInfo` to verify that product line item really has been added.